tests/unittests/media/server/gstplayer/genericPlayer/common/GenericTasksTestsBase.cpp

```diff
@@ -308,38 +315,100 @@ void GenericTasksTestsBase::expectSetupAudioElement()
     expectSetupVideoElement();
 }
 
 void GenericTasksTestsBase::shouldSetupVideoElementAmlhalasink()
 {
-    EXPECT_CALL(*testContext->m_glibWrapper, gStrHasPrefix(_, CharStrMatcher("westerossink"))).WillOnce(Return(false));
+    EXPECT_CALL(*testContext->m_glibWrapper, gTypeName(G_OBJECT_TYPE(testContext->m_element)))
+        .WillOnce(Return(kElementTypeName.c_str()));
     EXPECT_CALL(*testContext->m_glibWrapper, gStrHasPrefix(_, CharStrMatcher("amlhalasink"))).WillOnce(Return(true));
     EXPECT_CALL(*testContext->m_glibWrapper,
-                gObjectSetStub(G_OBJECT(&(testContext->m_element)), CharStrMatcher("wait-video")));
+                gObjectSetStub(G_OBJECT(testContext->m_element), CharStrMatcher("wait-video")));
     EXPECT_CALL(*testContext->m_glibWrapper,
-                gObjectSetStub(G_OBJECT(&(testContext->m_element)), CharStrMatcher("a-wait-timeout")));
+                gObjectSetStub(G_OBJECT(testContext->m_element), CharStrMatcher("a-wait-timeout")));
     EXPECT_CALL(*testContext->m_glibWrapper,
-                gObjectSetStub(G_OBJECT(&(testContext->m_element)), CharStrMatcher("disable-xrun")));
+                gObjectSetStub(G_OBJECT(testContext->m_element), CharStrMatcher("disable-xrun")));
+    expectSetupVideoElement();
+}
+
+void GenericTasksTestsBase::shouldSetupVideoElementPendingGeometryNotSupported()
+{
+    testContext->m_context.pendingGeometry = kRectangle;
+    EXPECT_CALL(*testContext->m_glibWrapper, gTypeName(G_OBJECT_TYPE(testContext->m_element)))
+        .WillOnce(Return(kElementTypeName.c_str()));
+    EXPECT_CALL(*testContext->m_glibWrapper, gStrHasPrefix(_, CharStrMatcher("amlhalasink"))).WillOnce(Return(false));
     expectSetupVideoElement();
 }
 
-void GenericTasksTestsBase::shouldSetupVideoElementPendingGeometryNonWesterissink()
+void GenericTasksTestsBase::shouldSetupVideoElementBrcmvideosink()
 {
     testContext->m_context.pendingGeometry = kRectangle;
-    EXPECT_CALL(*testContext->m_glibWrapper, gStrHasPrefix(_, CharStrMatcher("westerossink"))).WillOnce(Return(false));
+    EXPECT_CALL(*testContext->m_glibWrapper, gTypeName(G_OBJECT_TYPE(testContext->m_element)))
+        .WillOnce(Return(kBrcmSinkTypeName.c_str()));
     EXPECT_CALL(*testContext->m_glibWrapper, gStrHasPrefix(_, CharStrMatcher("amlhalasink"))).WillOnce(Return(false));
+    EXPECT_CALL(testContext->m_gstPlayer, setVideoSinkRectangle());
+    expectSetupVideoElement();
+}
+
+void GenericTasksTestsBase::shouldSetupVideoElementAutoVideoSinkWithChildren()
+{
+    expectSetupVideoElementAutoVideoSink(true);
+}
+
+void GenericTasksTestsBase::shouldSetupVideoElementAutoVideoSinkWithoutChildren()
+{
+    expectSetupVideoElementAutoVideoSink(false);
+}
+
+void GenericTasksTestsBase::shouldSetupVideoElementAutoVideoSinkWithMultipleChildren()
+{
+    testContext->m_iterator.size = 2;
+    expectSetupVideoElementAutoVideoSink(true);
+}
+
+void GenericTasksTestsBase::expectSetupVideoElementAutoVideoSink(bool hasChilden)
```

Comment:
Yeah your right, I think this code should be split up better to avoid that. It makes it harder to follow test code when you have logic in the test functions.

Reply:
Surely not too hard for someone who can write code? And keeping the boolean there makes it much easier to maintain the tests?

But, I can see that it might be easier for someone who doesn't write code to follow the test if there's no logic in there